File: app/src/main/java/com/doubean/ford/data/vo/PostCommentsResult.java
package com.doubean.ford.data.vo;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.List;

@Entity(tableName = "post_comments_results")
public class PostCommentsResult extends ListResult {
    @PrimaryKey
    @NonNull
    public String postId;

    public PostCommentsResult(@NonNull String postId, @Nullable List<String> ids, int totalCount, @Nullable Integer next) {
        super(ids, totalCount, next);
        this.postId = postId;
    }
}

File: app/src/main/java/com/doubean/ford/data/vo/PostCommentsResult.kt
package com.doubean.ford.data.vo

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "post_comments_results")
class PostCommentsResult(
    @PrimaryKey val postId: String,
    override val ids: List<String>,
    override val totalCount: Int,
    override val next: Int?
) : ListResult